lib/cloud.proxy.js

```diff
@@ -367,49 +369,50 @@ IoTKitCloud.prototype.dataSubmit = function (metric, callback) {
     var handler = function() {    
         // SparkplugB format data submission to sdk
         if (me.spBProxy != undefined) {
-            var componentMetrics = [];          
-            if (Array.isArray(metric)) {
-                metric.forEach(item => {
-                    let compMetric = me.spbMetricList.find(element => element.alias === item.alias);
-                    if (compMetric != undefined) {
-                        compMetric.value = item.v;
-                        compMetric.timestamp = item.on || new Date().getTime();
-                        componentMetrics.push(compMetric);
-                    }    
-                })
-                // Check if we have any data left to submit
-                if (componentMetrics.length === 0) {
-                    me.logger.error(' SPB Data submit - no data to submit.');
-                    return callback(false);
-                }
-            } else {
-                let compMetric = me.spbMetricList.find(element => element.alias === metric.alias);    
-                if (compMetric != undefined) {
-                    compMetric.value = metric.v;
-                    compMetric.timestamp = metric.on || new Date().getTime();
-                    componentMetrics.push(compMetric); 
+            let checkFlag = function () {
```

Comment:
I understand the need for this but I also have some concerns around the implementation:

- It feels kinda hacky polling endlessly for the birth message status
- What happens if the birth message is not completed? Do we just end up with potentially hundreds of threads looping round checking the birth message status?
- Given we probably need some kind of timeout provisions to cater for the birth message never completing what are the implications of just rejecting data until the birth message has been completed?

Reply:
I agree, I should add some timeout if the birth message status is not received as true, else it will be in loop.I will add some timeout mechanisam of closing data send function call when we want to send data.